core/go/internal/domainmgr/domain.go

```diff
@@ -771,3 +775,55 @@ func (d *domain) BuildDomainReceipt(ctx context.Context, dbTX *gorm.DB, txID uui
 	}
 	return tktypes.RawJSON(res.ReceiptJson), nil
 }
+
+func (d *domain) SendTransaction(ctx context.Context, tx *prototk.SendTransactionRequest) (*prototk.SendTransactionResponse, error) {
+	txType := pldapi.TransactionTypePrivate
+	if tx.Transaction.Type == prototk.TransactionInput_PUBLIC {
+		txType = pldapi.TransactionTypePublic
+	}
+	contractAddress, err := tktypes.ParseEthAddress(tx.Transaction.ContractAddress)
+	if err != nil {
+		return nil, err
+	}
+	var functionABI abi.Entry
+	if err = json.Unmarshal([]byte(tx.Transaction.FunctionAbiJson), &functionABI); err != nil {
+		return nil, err
+	}
+
+	id, err := d.dm.txManager.SendTransaction(ctx, &pldapi.TransactionInput{
+		TransactionBase: pldapi.TransactionBase{
+			Type: txType.Enum(),
+			From: tx.Transaction.From,
+			To:   contractAddress,
+			Data: tktypes.RawJSON(tx.Transaction.ParamsJson),
+		},
+		ABI: abi.ABI{&functionABI},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &prototk.SendTransactionResponse{Id: id.String()}, nil
+}
+
+func (d *domain) LocalNodeName(ctx context.Context, req *prototk.LocalNodeNameRequest) (*prototk.LocalNodeNameResponse, error) {
+	return &prototk.LocalNodeNameResponse{
+		Name: d.dm.transportMgr.LocalNodeName(),
+	}, nil
+}
+
+func (d *domain) GetStates(ctx context.Context, req *prototk.GetStatesRequest) (*prototk.GetStatesResponse, error) {
```

Comment:
I believe this (as implemented) is `GetConfirmedStatesByID`.

For some reason we pass it to the state context, but it doesn't use the state context. Would it make more sense to just call `statestore.FindStates(ctx, ss.p.DB(), domain, schema, &query, status)` directly?

Reply:
I believe what I want is `GetStatesByID`. I want to be able to look up the private data for a state, and depending on timing, the state may be proposed, confirmed, or spent by the time I perform the lookup.

Reply:
I've gone ahead and:
- Renamed to `GetStatesByID`
- Made it return un-flushed states from the Domain Context as well as anything in the DB